pact4s/src/test/scala/org/broadinstitute/dsde/rawls/provider/RawlsProviderSpec.scala

```diff
@@ -0,0 +1,311 @@
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.stream.Materializer
+import bio.terra.workspace.model.{
+  CloningInstructionsEnum,
+  DataRepoSnapshotAttributes,
+  DataRepoSnapshotResource,
+  ResourceMetadata,
+  ResourceType,
+  StewardshipType
+}
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import io.opentelemetry.context.Context
+import org.broadinstitute.dsde.rawls.TestExecutionContext.testExecutionContext
+import org.broadinstitute.dsde.rawls.billing.BillingProjectOrchestrator
+import org.broadinstitute.dsde.rawls.bucketMigration.BucketMigrationService
+import org.broadinstitute.dsde.rawls.dataaccess.{ExecutionServiceCluster, SamDAO}
+import org.broadinstitute.dsde.rawls.entities.EntityService
+import org.broadinstitute.dsde.rawls.genomics.GenomicsService
+import org.broadinstitute.dsde.rawls.model.{
+  ApplicationVersion,
+  RawlsRequestContext,
+  SnapshotListResponse,
+  StatusCheckResponse,
+  SubsystemStatus,
+  Subsystems,
+  UserInfo
+}
+import org.broadinstitute.dsde.rawls.snapshot.SnapshotService
+import org.broadinstitute.dsde.rawls.spendreporting.SpendReportingService
+import org.broadinstitute.dsde.rawls.status.StatusService
+import org.broadinstitute.dsde.rawls.user.UserService
+import org.broadinstitute.dsde.rawls.webservice.RawlsApiServiceImpl
+import org.broadinstitute.dsde.rawls.workspace.{MultiCloudWorkspaceService, WorkspaceService}
+import org.broadinstitute.dsde.workbench.oauth2.OpenIDConnectConfiguration
+import org.mockito.ArgumentMatchers.{any, anyInt, anyString}
+import org.mockito.Mockito.when
+import org.mockito.stubbing.OngoingStubbing
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.flatspec.AnyFlatSpec
+import pact4s.provider.Authentication.BasicAuth
+import pact4s.provider.{ConsumerVersionSelectors, PactSource, ProviderInfoBuilder, ProviderTags}
+import pact4s.scalatest.PactVerifier
+import pact4s.provider._
+import org.scalatest.time.SpanSugar.convertIntToGrainOfTime
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.typelevel.log4cats.StructuredLogger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import pact4s.provider.StateManagement.StateManagementFunction
+
+import java.lang.Thread.sleep
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.concurrent.duration.FiniteDuration
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
+import org.broadinstitute.dsde.rawls.model.Subsystems.Subsystem
+import org.broadinstitute.dsde.rawls.openam.MockUserInfoDirectives
+
+import java.io.File
+import scala.collection.immutable.Map
+
+object States {
+  val oneSnapshot = "one snapshot in the given workspace"
+  val rawlsOK = "Rawls is ok"
+}
+
+class RawlsProviderSpec extends AnyFlatSpec with BeforeAndAfterAll with PactVerifier {
+
+  private val WORKSPACE_UUID = "facade00-0000-4000-a000-000000000000"
```

Comment:
TODO: Can I pass these as state params or something?

Reply:
[ref](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java#L200-L212) This is how I did provider state stuff in WSM; maybe something similar is usable here?

Reply:
I've been mussing with it a bit, it's different for scala and I'm not confident I'll be able to get it to work.